libsplinter/src/circuit/template/yaml_parser/mod.rs

```diff
@@ -45,16 +45,36 @@ impl CircuitTemplate {
     ///
     /// * `file_path` - Path to the template YAML file.
     pub fn load_from_file(file_path: &str) -> Result<Self, CircuitTemplateError> {
-        let path = Path::new(file_path);
-        if !path.is_file() {
-            return Err(CircuitTemplateError::new(&format!(
-                "File does not exist or is inaccessible: {}",
-                file_path
-            )));
-        }
-        let file = File::open(path).map_err(|err| {
-            CircuitTemplateError::new_with_source("Error opening template file", err.into())
-        })?;
+        let custom_path = Path::new(file_path);
```

Comment:
Take a look at https://github.com/hyperledger/transact/blob/master/libtransact/src/contract/archive/mod.rs for an example of searching a path for a file.